api/v1alpha1/controlplane_types.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+type GenericControlPlaneSpec struct {
+	// AutoRenewCertificates specifies the configuration for auto-renewing the
+	// certificates of the control plane.
+	// +kubebuilder:validation:Optional
+	AutoRenewCertificates *AutoRenewCertificatesSpec `json:"autoRenewCertificates,omitempty"`
+}
+
+type AutoRenewCertificatesSpec struct {
+	// DaysBeforeExpiry indicates a rollout needs to be performed if the
+	// certificates of the control plane will expire within the specified days.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=7
```

Comment:
~Is thins minimum coming from CAPI?~ Yes https://cluster-api.sigs.k8s.io/tasks/certs/auto-rotate-certificates-in-kcp